app/src/main/java/org/sopt/dosopttemplate/presentation/login/LoginActivity.kt

```diff
@@ -0,0 +1,96 @@
+import org.sopt.dosopttemplate.databinding.ActivityLoginBinding
+import org.sopt.dosopttemplate.presentation.main.MainActivity
+import org.sopt.dosopttemplate.presentation.signup.SignupActivity
+import org.sopt.dosopttemplate.util.activity.hideKeyboard
+import org.sopt.dosopttemplate.util.binding.BindingActivity
+import org.sopt.dosopttemplate.util.context.toast
+import org.sopt.dosopttemplate.util.intent.getParcelable
+import org.sopt.dosopttemplate.util.view.UiState
+import org.sopt.dosopttemplate.util.view.snackBar
+
+@AndroidEntryPoint
+class LoginActivity : BindingActivity<ActivityLoginBinding>(R.layout.activity_login) {
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent>
+    private val viewModel by viewModels<LoginViewModel>()
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding.vm = viewModel
+        initResultLauncher()
+        initSignupButtonClickListener()
+        initLoginButtonClickListener()
+        initloginStateObserver()
+        initHideKeyboard()
+    }
+
+    private fun initResultLauncher() {
+        resultLauncher = registerForActivityResult(
+            ActivityResultContracts.StartActivityForResult()
+        ) { result ->
+            if (result.resultCode == RESULT_OK) {
+                val user = result.data?.getParcelable(USER_KEY, User::class.java)
+                viewModel.setAutoLogin(user)
+                binding.root.snackBar { getString(R.string.login_success_signup) }
+                setLoginButtonClickListener(user)
+            }
+        }
+    }
+
+    private fun initSignupButtonClickListener() {
+        binding.btnLoginSignup.setOnClickListener {
+            val intent = Intent(this, SignupActivity::class.java)
+            resultLauncher.launch(intent)
+        }
+    }
+
+    private fun initLoginButtonClickListener() {
+        binding.btnLoginLogin.setOnClickListener {
+            viewModel.login()
+        }
+    }
+
+    private fun setLoginButtonClickListener(user: User?) {
+        binding.btnLoginLogin.setOnClickListener {
+            viewModel.login(user)
+        }
+    }
+
+    private fun initloginStateObserver() {
+        viewModel.loginState.observe(this) { state ->
+            when (state) {
+                is UiState.Success -> {
+                    navigateToMainScreenWithUserData(state.data)
+                }
+
+                is UiState.Empty -> {}
+                else -> binding.root.snackBar { getString(R.string.login_fail_login) }
+            }
+        }
+    }
+
+    private fun navigateToMainScreenWithUserData(user: User?){
+        val intent = Intent(this, MainActivity::class.java)
+        intent.putExtra(USER_KEY, user)
+        this.toast(getString(R.string.login_success_login))
+        viewModel.setAutoLogin(user)
+        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+        startActivity(intent)
+    }
```

Comment:
intent에 .apply{} 로 한번에 연결된 코드를 만들면 더 깔끔한 코드가 될 것 같습니다 !
```suggestion
this.toast(getString(R.string.login_success_login))
viewModel.setAutoLogin(user)
Intent(this, MainActivity::class.java).apply{
        putExtra(USER_KEY, user)
        addFlags(FLAG_ACTIVITY_NEW_TASK or FLAG_ACTIVITY_CLEAR_TASK)
        startActivity(this)
}
```